graylog2-server/src/main/java/org/graylog2/decorators/LinkFieldDecorator.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.decorators;
+
+import com.floreysoft.jmte.Engine;
+import com.floreysoft.jmte.template.Template;
+import com.floreysoft.jmte.template.VariableDescription;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.assistedinject.Assisted;
+import org.graylog2.plugin.Message;
+import org.graylog2.plugin.configuration.ConfigurationRequest;
+import org.graylog2.plugin.configuration.fields.BooleanField;
+import org.graylog2.plugin.configuration.fields.TextField;
+import org.graylog2.plugin.decorators.SearchResponseDecorator;
+import org.graylog2.rest.models.messages.responses.ResultMessageSummary;
+import org.graylog2.rest.resources.search.responses.SearchResponse;
+
+import javax.inject.Inject;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+public class LinkFieldDecorator implements SearchResponseDecorator {
+
+    private static final String CK_LINK_FIELD = "link_field";
+
+    private final String linkField;
+
+    public interface Factory extends SearchResponseDecorator.Factory {
+        @Override
+        LinkFieldConverter create(Decorator decorator);
```

Comment:
There are a bunch of the old `LinkFieldConverter` class name that must be renamed here and in `DecoratorBindings`.